components/brave_ads/browser/buildflags/BUILD.gn

```diff
@@ -1,9 +1,10 @@
-import("//build/buildflag_header.gni")
 import("//brave/components/brave_ads/browser/buildflags/buildflags.gni")
+import("//build/buildflag_header.gni")
 
 buildflag_header("buildflags") {
   header = "buildflags.h"
   flags = [
     "BRAVE_ADS_ENABLED=$brave_ads_enabled",
+    "ENABLE_NATIVE_NOTIFICATIONS=$enable_native_notifications",
```

Comment:
I believe this was an upstream flag so is it still needed? cc @tmancey

Reply:
This flag is still used in `src/brave/browser/brave_browser_process_impl.cc`. However, it will be removed once custom ad notifications ride the trains

Reply:
I think we need some other changes then @mkarolin because if this isn't an upstream buildflag anymore then stuff like this is no longer correct
```#if BUILDFLAG(ENABLE_NATIVE_NOTIFICATIONS)
#include "chrome/browser/notifications/notification_platform_bridge.h"
#include "brave/browser/notifications/brave_notification_platform_bridge.h"
#endif``` from brave_browser_process_impl.cc

Reply:
in fact that was never correct because brave_notification_platform_bridge is not behind that buildflag in gn

Reply:
and this is also a problem ```NotificationPlatformBridge*
BraveBrowserProcessImpl::notification_platform_bridge() {
#if !defined(OS_MAC)
  return BrowserProcessImpl::notification_platform_bridge();
#else
#if BUILDFLAG(ENABLE_NATIVE_NOTIFICATIONS)
  if (!created_notification_bridge_)
    CreateNotificationPlatformBridge();
  return notification_bridge_.get();
#else
  return nullptr;
#endif
#endif
}

void BraveBrowserProcessImpl::CreateNotificationPlatformBridge() {
#if defined(OS_MAC)
#if BUILDFLAG(ENABLE_NATIVE_NOTIFICATIONS)
  DCHECK(!notification_bridge_);
  notification_bridge_ = BraveNotificationPlatformBridge::Create();
  created_notification_bridge_ = true;
#endif
#endif
}```

Reply:
if the guard is gone in upstream, not upstream code should be behind it. Also `BraveNotificationPlatformBridge` was never behind `ENABLE_NATIVE_NOTIFICATIONS` in gn and should not be behind it now either

Reply:
and actually I don't think this buildflag works at all anymore because the upstream code no longer uses it

Reply:
@tmancey, PTAL at the changes here: https://github.com/brave/brave-core/pull/7866/commits/f4c17ac107b76c22862e048a10a1b98352bdd9c8